idea/src/org/jetbrains/kotlin/idea/inspections/MemberVisibilityCanPrivateInspection.kt

```diff
@@ -112,9 +112,18 @@ class MemberVisibilityCanPrivateInspection : AbstractKotlinInspection() {
             else -> "Property"
         }
         val nameElement = (declaration as? PsiNameIdentifierOwner)?.nameIdentifier ?: return
-        holder.registerProblem(declaration.visibilityModifier() ?: nameElement,
+        val visibilityModifier = declaration.visibilityModifier()
+        val fix = object : AddModifierFix(modifierListOwner, KtTokens.PRIVATE_KEYWORD) {
```

Comment:
I would do this as a general fix to AddModifierFix (which already has special logic for handling modality modifiers), not as a single-case override here.